### PR TITLE
Reorganize project structure, moving TSS2 implementation to sub-project

### PR DESCRIPTION
**Pull request by zanebeckwith**
This PR moves what previously was the bulk of this project (an implementation of a subset of the TCG TSS2.0 stack) into a sub-project (in the directory `tss2`) and promotes what previously was a small utility (`nvram`) to the main project.

This means the `libxaptum-tpm` library contains (currently) just the `nvram` utility functions, while the new `libtss2-[sys,tcti-device,tcti-mssim]` libraries contain the old TSS2 functions.

The TSS2 implementation is also updated to use the naming scheme of the current version of the `tpm2-software` project's implementation of the TSS, so that any project that requires only the subset implemented by this project should be able to use this implementation or the `tpm2-software` one interchangeably (for example, I've been able to build the new `ecdaa` project against this new TSS2).

The future plan is to add new functionality (i.e. key generation) to the `libxaptum-tpm` project, and leave the `libtss2-*` libraries available in case any usecase desires to use a minimal TSS implementation

Also, in the Travis-CI builds, the main project now builds by default against the `tpm2-software` implementation of the TSS (i.e. we build a local installation of that library). This PR adds a new job to build our own `libtss2-*` libraries, test them, and build the main project against them, as well.

### CODE REVIEW

**Comment by zanebeckwith** [comment]
The failing Travis build is due to the issue mentioned in PR #55 . Once that PR is merged, I'll rebase this branch to obtain that fix, and the CI *should* then succeed.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2017 Xaptum, Inc.
+# Copyright 2020 Xaptum, Inc.
```
Comment (drbild) [inline test/CMakeLists.txt:1]: Nit pick: use a range here to document the age of the project:
```
# Copyright 2017-2020 Xaptum, Inc.
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -86,6 +74,16 @@ if(BUILD_SHARED_LIBS)
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   )
 
+  if(BUILD_TSS2)
+    target_link_libraries(xaptum-tpm PUBLIC
+      tss2-sys
```
Comment (drbild) [inline CMakeLists.txt:79]: Can the same `tss2::sys`  syntax be used in both cases?   Maybe just add an `tss2::sys` alias in our `CMakeLists.txt`, same as is in the `FindTSS2.cmake` script.

File: test/CMakeLists.txt (hunk 1 above)
Comment (zanebeckwith) [reply test/CMakeLists.txt:1]: Yes, good catch

File: CMakeLists.txt (hunk 2 above)
Comment (zanebeckwith) [reply CMakeLists.txt]: Ha, I actually originally did have that (the library alias), but worried that it actually made the static library case too messy

I just pushed a variant with this change

Comment (zanebeckwith) [reply CMakeLists.txt]: Sorry @drbild I didn't notice a naming problem that made the CI fail. Fixed now

**Pull request merged**